### ISSUE TITLE
ftol and xtol for _minimize_neldermead are absolute instead of relative

### ISSUE THREAD

**Issue by tanriol** [scipy/scipy#5051]
The docstring says that
```
    xtol : float
        Relative error in solution `xopt` acceptable for convergence.
    ftol : float
        Relative error in ``fun(xopt)`` acceptable for convergence.
```

But the loop is left through
```python
        if (numpy.max(numpy.ravel(numpy.abs(sim[1:] - sim[0]))) <= xtol and
                numpy.max(numpy.abs(fsim[0] - fsim[1:])) <= ftol):
            break
```
which seems to be a condition on absolute errors instead of relative.

**Comment by argriffing** [comment]
It is an absolute error, but I think it's an absolute error between iterations (as opposed to the difference between the measured or inferred value of a quantity x_0 and its actual value x). The analogous Matlab descriptions are just `Termination tolerance on x` and `Termination tolerance on the function value`. I'm not sure what's the best terminology.

**Comment by pv** [comment]
The options then should be `xatol` and `fatol`. Needs deprecation cycle.

**Comment by andyfaff** [comment]
Fixed in #5939
